src/amo/reducers/currentView.js

```diff
@@ -0,0 +1,38 @@
+/* @flow */
+import { CURRENT_VIEW_SET } from 'core/constants';
+
+
+export type currentViewTypes = {|
+  // TODO: Create an addonType Type that is one of type (ADDON_TYPE_EXTENSION,
+  // ADDON_TYPE_THEME, etc.)
+  addonType: string | null,
+  isExploring: boolean | void,
+  isHomePage: boolean | void,
+|}
+
+export type currentViewActionType = {|
+  type: string,
+  payload: any,
+|}
+
+export const initialState = {
+  addonType: null,
+  isExploring: false,
+  isHomePage: false,
+};
+
+export default function currentView(
+  state : currentViewTypes = initialState, action : currentViewActionType
+) {
+  const { payload } = action;
+  switch (action.type) {
+    case CURRENT_VIEW_SET:
```

Comment:
To me, `CURRENT_VIEW_CONTEXT` would make more sense because this state describes the context in which one is viewing add-ons. I think you could also drop *current* since that's implied by the persistent nature of the redux state, making it `VIEW_CONTEXT`.